dashboard/src/main/home/infrastructure/components/InfraResourceList.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Context } from "shared/Context";
+import api from "shared/api";
+import styled from "styled-components";
+import Loading from "components/Loading";
+import { TFState } from "shared/types";
+
+type Props = {
+  infra_id: number;
+};
+
+const InfraResourceList: React.FunctionComponent<Props> = ({ infra_id }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [infraState, setInfraState] = useState<TFState>(null);
+  const { currentProject, setCurrentError } = useContext(Context);
+
+  useEffect(() => {
+    api
+      .getInfraState(
+        "<token>",
+        {},
+        {
+          project_id: currentProject.id,
+          infra_id: infra_id,
+        }
+      )
+      .then(({ data }) => {
+        setInfraState(data);
+
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        setHasError(true);
+        setCurrentError(err.response?.data?.error);
+        setIsLoading(false);
+      });
+  }, [currentProject]);
+
+  if (isLoading) {
+    return (
+      <Placeholder>
+        <Loading />
+      </Placeholder>
+    );
+  }
+
+  if (!infraState) {
+    return <Placeholder>No resources available</Placeholder>;
+  }
+
+  if (hasError) {
+    return <Placeholder>Error</Placeholder>;
+  }
+
+  console.log(infraState);
+
+  const renderContents = () => {
+    return Object.keys(infraState.resources).map((key) => {
```

Comment:
Always make sure that you're passing an object, if by any change infraState.resources is filled with anything besides an object this will directly throw an object.
A simple solution would be to use:
```suggestion
    return Object.keys({...(infraState?.resources || {})}).map((key) => {
```